### DISCUSSION TITLE
How to clear the swap.vhdx in the temp files

### DISCUSSION THREAD

**Discussion by JannikSchneider12** [microsoft/WSL#10885] [General]
Hey,

I am using WSL2 Ubuntu and I know that there is an issue with wsl in general capturing memory and not releasing it back. Thats why I wanted to use wslcompact but before that it says that I should clear my temp folder at first. The swap.vhdx file takes almost 50gb and I can not clear it. Does someone know how I can do this?

Thanks for your help

**Comment by paramidon** [comment]
Hi, this is not answer, but another complaint.

I do use use default TEMP values:
![image](https://github.com/microsoft/WSL/assets/55555207/b5cbc6f8-c10b-483c-bb53-c7c655769d29)

But WSL2 still uses OLD location. WTF? Why this buggy windows/wsl ignores parameters? Why these parameters exists if its ignored??

![image](https://github.com/microsoft/WSL/assets/55555207/b471b156-652d-48fd-a1e7-649c26a3b9df)

**Comment by ZagroCore** [comment]
Press Win+ R then  type: **_wsl --shutdown_**

**Comment by nicotie** [comment]
swap file is _'an absolute Windows path to the swap virtual hard disk.'_ Its default location is _%USERPROFILE%\AppData\Local\Temp\swap.vhdx._
To compact and redirect this filepath, i added the following in the .wslconfig
```
[wsl2]
# Sets swapfile path location, default is %USERPROFILE%\AppData\Local\Temp\swap.vhdx
# DEFAULT SETTING: swapfile=C:\\temp\\wsl-swap.vhdx
swapfile=E:\\wslswapfile\\wsl-swap.vhdx
```
**After redirecting and restart the wsl, the swap file in the disk C can be removed.**
But im not sure what will happen if the old file _swap.vhdx_ in the disk C were removed directly. Maybe this can be solved more appropriately by using _diskpart_ and compact command.

**Reply by nicotie** [reply]
here is the reference:
https://learn.microsoft.com/en-us/windows/wsl/wsl-config
